lib/tasks/ont_runs.rake

```diff
@@ -0,0 +1,100 @@
+    mutation($barcode: String!, $wells: [WellWithSamplesInput!]!) {
+      createPlateWithOntSamples(
+        input: {
+          arguments: {
+            barcode: $barcode
+            wells: $wells
+          }
+        }
+      ) { errors }
+    }
+  GRAPHQL
+
+  # Methods to create variable objects for GraphQL
+  class Variables
+    def wells(samples_per_well:)
+      well_positions = ((1..12).to_a.product %w[A B C D E F G H]).map { |pair| "#{pair[1]}#{pair[0]}" }
+
+      well_positions.map do |position|
+        well position: position, num_samples: samples_per_well
+      end
+    end
+
+    private
+
+    def sample(position:, sample_number:, tag_group_id_prefix:)
+      padded_sample_number = format('%<number>03i', { number: sample_number })
+
+      sample = {
+        'name' => "Sample #{padded_sample_number} for #{position}",
+        'externalId' => "#{position}-#{padded_sample_number}-ExtId"
+      }
+
+      unless tag_group_id_prefix.nil?
+        padded_tag_number = format('%<number>02i', { number: sample_number })
+        sample['tagGroupId'] = "#{tag_group_id_prefix}#{padded_tag_number}"
+      end
+
+      sample
+    end
+
+    def well(position:, num_samples:)
+      {
+        'position' => position,
+        'samples' => (1..num_samples).map do |number|
+          sample(
+            position: position,
+            sample_number: number,
+            tag_group_id_prefix: num_samples == 96 ? 'ont_96_tag_' : nil
+          )
+        end
+      }
+    end
+  end
+end
+
+namespace :ont_runs do
+  def show_errors(error_lines)
+    error_lines.each { |line| puts line }
+    exit
+  end
+
+  def submit_create_plate_query(description:, barcode:, wells:)
+    puts "-> Creating a plate for #{description}"
+    result = TractionGraphQL::Client.query(OntPlates::CreatePlate, variables: { barcode: barcode, wells: wells })
+
+    errors_array = result.original_hash['data']['createPlateWithOntSamples']['errors']
+    if errors_array.any?
+      show_errors ["-> Failed to create plate for #{description}: #{errors_array}"]
+    end
+
+    puts "-> Succesfully created a plate for #{description}"
+  rescue Errno::ECONNREFUSED
+    show_errors ["-> Failed to connect to the Rails server at #{TractionGraphQL::RAILS_ROOT_URI}",
+                 '   Use the RAILS_ROOT_URI environment variable to specify a different URI']
+  end
+
+  task create: :environment do
+    puts '-> Creating ONT runs using GraphQL'
+    puts '   Note these could take a few minutes to complete'
+    variables = OntPlates::Variables.new
+    submit_create_plate_query(description: 'pooling scenario 1', barcode: 'PLATE-PS01', wells: variables.wells(samples_per_well: 1))
+    submit_create_plate_query(description: 'pooling scenario 2', barcode: 'PLATE-PS02', wells: variables.wells(samples_per_well: 1))
+    submit_create_plate_query(description: 'pooling scenario 3', barcode: 'PLATE-PS03', wells: variables.wells(samples_per_well: 96))
```

Comment:
I'd argue that these are 'plate input scenarios'. And I thought the 3 scenarios would be a plate with 1 sample in a well; 96 samples in a well; and 384 samples in a well?

Reply:
Not sure what you mean.  The pooling scenarios require a plate of a specific set up.  Steve said not to worry about the 384 samples in a well.  The first two scenarios use a plate with 96 wells and a single sample in each.  The third scenario requires a plate with 96 wells, each containing either 96 or 384 samples.  We're just sticking to 96 for now.

Reply:
What I'm getting at is that at the plate creation stage there is no distinction between scenarios 1 and 2. They both create a plate with one sample in a well. But I suppose you're setting up separate plates for the different scenarios, so require setting up plates for those scenarios